agent/structs/config_entry_sameness_group.go

```diff
@@ -20,6 +20,17 @@ type SamenessGroupConfigEntry struct {
 	RaftIndex
 }
 
+func (e *SamenessGroupConfigEntry) ModifyID() string {
+	return e.Meta[ConfigEntryIDKey]
+}
+
+func (e *SamenessGroupConfigEntry) SetModifyID(id string) {
+	if e.Meta == nil {
+		e.Meta = make(map[string]string)
+	}
+	e.Meta[ConfigEntryIDKey] = id
```

Comment:
:golf:  You could do one generic function in `lib/maps` and make the modify of all of the `SetModifyID` functions be one liners instead:
```
func setSafely[K comparable, V any](m map[K]V, k K, v V) map[K]V {
	if m == nil {
		return map[K]V{k: v}
	}
	m[k] = v
	return m
}
```

Reply:
I will probably make this a separate field anyway based on @mkeeler comment.